pkgs/applications/networking/p2p/qbittorrent/default.nix

```diff
@@ -1,44 +1,47 @@
-{ stdenv, fetchurl, pkgconfig, which
+{ stdenv, fetchFromGitHub, pkgconfig
 , boost, libtorrentRasterbar, qtbase, qttools, qtsvg
 , debugSupport ? false # Debugging
-, guiSupport ? true, dbus_libs ? null # GUI (disable to run headless)
+, guiSupport ? true, dbus ? null # GUI (disable to run headless)
 , webuiSupport ? true # WebUI
 }:
 
-assert guiSupport -> (dbus_libs != null);
-with stdenv.lib;
+assert guiSupport -> (dbus != null);
 
 stdenv.mkDerivation rec {
   name = "qbittorrent-${version}";
-  version = "4.1.0";
+  version = "4.1.1";
 
-  src = fetchurl {
-    url = "mirror://sourceforge/qbittorrent/${name}.tar.xz";
-    sha256 = "0fdr74sc31x421sb69vlgal1hxpccjxxk8hrrzz9f5bg4jv895pw";
+  src = fetchFromGitHub {
+    owner = "qbittorrent";
+    repo = "qbittorrent";
+    rev = "release-${version}";
+    sha256 = "09bf1jr2sfdps8cb154gjw7zhdcpsamhnfbgacdmkfyd7qgcbykf";
   };
 
-  nativeBuildInputs = [ pkgconfig which ];
+  nativeBuildInputs = [ pkgconfig ];
 
-  buildInputs = [ boost libtorrentRasterbar qtbase qttools qtsvg ]
-    ++ optional guiSupport dbus_libs;
+  buildInputs = [ boost libtorrentRasterbar qtbase qttools qtsvg
+    (if guiSupport then dbus else [] ) # D(esktop)-Bus depends on GUI support
+  ];
 
   # Otherwise qm_gen.pri assumes lrelease-qt5, which does not exist.
   QMAKE_LRELEASE = "lrelease";
 
   configureFlags = [
     "--with-boost-libdir=${boost.out}/lib"
     "--with-boost=${boost.dev}"
-    (if guiSupport then "" else "--disable-gui")
+    (if guiSupport then "" else "--disable-gui --enable-systemd") # Also place qbittorrent-nox systemd service files
     (if webuiSupport then "" else "--disable-webui")
-  ] ++ optional debugSupport "--enable-debug";
+    (if debugSupport then "--enable-debug" else "")
+  ];
 
   enableParallelBuilding = true;
 
-  meta = {
-    description = "Free Software alternative to µtorrent";
+  meta = with stdenv.lib; {
+    description = "Featureful free software BitTorrent client";
     homepage    = https://www.qbittorrent.org/;
     license     = licenses.gpl2;
-    platforms   = platforms.linux;
+    platforms   = with platforms; unix;
```

Comment:
We should not extend the list of supported platforms without testing that they are actually supported, otherwise we will likely just burden macOS Hydra builders with an always broken build. I have asked GrahamcOfBorg to build this for Mac too; if it does not report success and no one else confirms that this builds on Darwin, the platforms should be restricted back to Linux.

Reply:
That is why I do it so.
I also do not have Darwin. So I look at bot build.

Reply:
The bot has attempted to build qbittorrent on Darwin and it did not succeed: https://github.com/NixOS/nixpkgs/pull/41252#issuecomment-393368796, https://logs.nix.ci/?key=nixos/nixpkgs.41252&attempt_id=ca7f7512-a616-42ca-bb60-f50432dc9303 . Supporting Darwin requires additional work, and it is out of scope of this PR.